service-compiler/src/main/java/io/techery/janet/HttpHelpersGenerator.java

```diff
@@ -57,12 +57,18 @@
 
     @Override
     public void generate(ArrayList<HttpActionClass> actionClasses) {
-        Set<TypeElement> checkSet = new HashSet<TypeElement>();
+        Set<TypeElement> processedElements = new HashSet<TypeElement>();
         for (HttpActionClass actionClass : actionClasses) {
-            while (actionClass != null
-                    && !checkSet.contains(actionClass.getTypeElement())) {
-                generate(actionClass);
-                checkSet.add(actionClass.getTypeElement());
+            while (actionClass != null && !processedElements.contains(actionClass.getTypeElement())) {
+                // generate only if there is no such helper class yet
+                try {
+                    String name = actionClass.getHelperName().packageName() +
+                            "." + actionClass.getHelperName().simpleName();
+                    Class.forName(name, false, getClass().getClassLoader());
```

Comment:
Sometimes  we want to get some information about some element or type. 
Maybe we should put this information into `HttpActionClass`. What do you think about it?